cpp/src/aztec/honk/pcs/gemini/gemini.test.cpp

```diff
@@ -0,0 +1,156 @@
+#include "gemini.hpp"
+
+#include "../commitment_key.test.hpp"
+#include <gtest/gtest.h>
+
+namespace honk::pcs::gemini {
+
+template <class Params> class GeminiTest : public CommitmentTest<Params> {};
+
+TYPED_TEST_SUITE(GeminiTest, CommitmentSchemeParams);
+
+TYPED_TEST(GeminiTest, single)
+{
+    using Gemini = MultilinearReductionScheme<TypeParam>;
+    using MLEOpeningClaim = MLEOpeningClaim<TypeParam>;
+
+    const size_t n = 16;
+    const size_t m = 4; // = log(n)
+
+    auto u = this->random_evaluation_point(m);
+    auto poly = this->random_polynomial(n);
+    auto commitment = this->commit(poly);
+    auto eval = poly.evaluate_mle(u);
+
+    // create opening claim
+    auto claims = { MLEOpeningClaim{ commitment, eval, false } };
+
+    this->consume(u);
+
+    auto [prover_claim, witness, proof] =
+        Gemini::reduce_prove(this->ck(), u, claims, { &poly }, this->prover_challenges);
```

Comment:
I'm more inclined to keep the term `reduce`, since we're reducing both the claims and the witness, and constructing a proof. 
However, I agree that `reduce_prove` isn't super descriptive. 
I'm thinking something like `reduce_to_batch_univariate_claim` ?